intellij-plugin-verifier/verifier-intellij/src/main/java/com/jetbrains/pluginverifier/usages/overrideOnly/OverrideOnlyMethodUsageProcessor.kt

```diff
@@ -6,37 +6,100 @@ package com.jetbrains.pluginverifier.usages.overrideOnly
         OverrideOnlyMethodUsage(methodReference, resolvedMethod.location, callerMethod.location)
       )
     }
   }
 
-  private fun isCallOfSuperConstructor(callerMethod: Method, resolvedMethod: Method) =
-    resolvedMethod.isConstructor
-      && callerMethod.isConstructor
-      && callerMethod.containingClassFile.superName == resolvedMethod.containingClassFile.name
+  /**
+   * Detects if this is an allowed method invocation.
+   * In other words, if this is an allowed scenario where the invocation of such method is allowed and will not
+   * be reported as a plugin problem.
+   *
+   * Example:
+   * ```
+   *   public void usages(OverrideOnlyMethodOwner owner1,) {
+   *     owner1.overrideOnlyMethod();
+   *   }
+   * ```
+   * Description:
+   * - Both _Invoked Method Reference_ and _Invoked Method_ will refer to the `overrideOnlyMethod`.
+   * - _Invocation Instruction_ will refer to the JVM bytecode instruction which will invoke the `overrideOnlyMethod`.
+   * In this sample, this will be `182/invokevirtual` opcode.
+   * - _Caller Method_ refers to the `usages()` method
+   * @param invokedMethodReference a reference to the `OverrideOnly` method that is being invoked
+   * @param invokedMethod the `OverrideOnly` method that is being invoked
+   * @param invocationInstruction low-level JVM invocation instruction used to invoke the `OverrideOnly` method
+   * @param callerMethod the method in which the invocation of `OverrideOnly` method occurs.
+   * @param context the verification context with additional metadata and data
+   *
+   */
+  private fun isAllowedOverrideOnlyUsage(invokedMethodReference: MethodReference,
+                                         invokedMethod: Method,
+                                         invocationInstruction: AbstractInsnNode,
+                                         callerMethod: Method,
+                                         context: VerificationContext): Boolean {
+    return allowedOverrideOnlyUsageFilter.allowMethodInvocation(invokedMethod, invocationInstruction, callerMethod, context)
+  }
 
-  private fun Method.isOverrideOnlyMethod(): Boolean =
+  private fun Method.isOverrideOnlyMethod(context: VerificationContext): Boolean =
     annotations.findAnnotation(overrideOnlyAnnotationName) != null
       || containingClassFile.annotations.findAnnotation(overrideOnlyAnnotationName) != null
+      || isAnnotationPresent(overrideOnlyAnnotationName, context)
+
+  private fun Method.isAnnotationPresent(annotationFqn: String, verificationContext: VerificationContext): Boolean {
+    if (findEffectiveMemberAnnotation(annotationFqn, verificationContext.classResolver) != null) {
+      return true
+    }
+
+    val overriddenMethod = searchParentOverrides(verificationContext.classResolver).firstOrNull { (overriddenMethod, c) ->
+       overriddenMethod.findEffectiveMemberAnnotation(annotationFqn, verificationContext.classResolver) != null
+    }
+    return if (overriddenMethod == null) {
```

Comment:
should we use LOG.isLEVELEnabled() here for performance

Reply:
Fixed.